src/main/java/org/asciidoc/intellij/lexer/asciidoc.flex

```diff
@@ -400,19 +400,21 @@ LINKEND = "]"
   [^]                  { return AsciiDocTokenTypes.REF; }
 }
 
-<LINKFILE> {
-  "#"                  { yybegin(LINKANCHOR); return AsciiDocTokenTypes.LINKANCHOR; }
+<LINKFILE, LINKANCHOR> {
```

Comment:
@bbrenne - this is a refactoring; both states LINKFILE and LINKANCHOR watch for LINKTEXT_START. This is can be combined like this.